### PR TITLE
Documentation for SSL and setup simplification

### PR DESCRIPTION
**Pull request by aug24**
## What does this change?
Small incremental improvements to the install process: move common elements of configuration into common config.

Add a troubleshooting document for ssl issues caused by missing cert or pem.

## How can success be measured?
No change to production services.

## Tested?
- [X] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment

### CODE REVIEW

File: usage/app/lib/UsageConfig.scala [hunk 1]
```diff
@@ -64,8 +64,8 @@ class UsageConfig(resources: GridConfigResources) extends CommonConfig(resources
     try {
       iamClient.getUser.getUser.getUserName
     } catch {
-      case e:com.amazonaws.AmazonServiceException=>
-        logger.warn("Unable to determine current IAM user, probably because you're using temp credentials.  Usage may not be able to determine the live/preview app names")
+      case e:com.amazonaws.SdkClientException =>
```
Comment (aug24) [inline usage/app/lib/UsageConfig.scala:67]: This is the parent exception to AmazonServiceException - this change captures the client side error as well as server error responses.

File: docs/04-troubleshooting/04-ssl-certs.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+GRID services make HTTPS calls between themselves, and need a root certificate to do so.
+
+## Generating certificate.
+
+Installation is trivial:
+
+Ensure JAVA_HOME is exported, and then run
+```
+mkcert -install
```
Comment (sihil) [inline docs/04-troubleshooting/04-ssl-certs.md:11]: Is this not done as part of `setup.sh`?

File: dev/script/generate-config/service-config.js [hunk 3]
```diff
@@ -188,12 +183,14 @@ module.exports = {
...
         return {
             auth: getAuthConfig(config),
+            'admin-tools': getAdminToolsConfig(config),
             collections: getCollectionsConfig(config),
             cropper: getCropperConfig(config),
             'image-loader': getImageLoaderConfig(config),
             kahuna: getKahunaConfig(config),
             leases: getLeasesConfig(config),
             'media-api': getMediaApiConfig(config),
+            // 'media-usage': getUsageConfig(config),
```
Comment (sihil) [inline dev/script/generate-config/service-config.js:193]: delete if not used?

File: dev/script/generate-config/service-config.js [hunk 4]
```diff
@@ -12,28 +12,31 @@ function getCorsAllowedOriginString(config) {
 function getCommonConfig(config) {
   return `domain.root="${config.DOMAIN}"
         |authentication.providers.machine.config.authKeyStoreBucket="${config.coreStackProps.KeyBucket}"
+        |aws.local.endpoint="https://localstack.media.${config.DOMAIN}"
```
Comment (sihil) [inline dev/script/generate-config/service-config.js:15]: nice DRYing out....

File: dev/script/start.sh [hunk 5]
```diff
@@ -92,6 +92,9 @@ buildJs() {
 
 startPlayApps() {
   # pushd to find build.sbt and allow this script to be executed from any location (but ideally from the project root)
+  echo "========================================================="
+  echo "= Press cmd-C then cmd-D to stop"
+  echo "========================================================="
```
Comment (sihil) [inline dev/script/start.sh:97]: Ah TIL. 💯

**Review by sihil — APPROVED** [review]
Seems sane. Couple of thoughts.

File: docs/04-troubleshooting/04-ssl-certs.md (hunk 2 above)
Comment (akash1810) [reply docs/04-troubleshooting/04-ssl-certs.md:11]: This step is performed by [`dev-nginx`](https://github.com/guardian/dev-nginx/blob/e41c9afcea2c30f2d9c92b980c6f3c1ecaa5531d/script/setup-cert#L48), which `setup.sh` calls out to; I don't think this is needed here.

Comment (aug24) [reply docs/04-troubleshooting/04-ssl-certs.md:11]: Will move to a rectification section below.

File: dev/script/start.sh (hunk 5 above)
Comment (AWare) [reply dev/script/start.sh:97]: I thought it was ^C ^D not ⌘C ⌘D?

**Pull request merged**